main.go

```diff
@@ -0,0 +1,5 @@
+package main
+
+func main() {
```

Comment:
Obviously this is just a placeholder, but when you do create a main function, you'll probably want to put it in a different package (maybe a subdir called `cmd/<command name>` or something like that) and put that in a main package.

Reply:
I'm curious why I would want to put a command into a different package.  Wouldn't it be ideal if I could import the library and use it as a library and the same package when built on its own would be a CLI tool?  I'm probably missing something about how go programs are organized.

Reply:
That would be nice, But unfortunately in that case, all calling could would need to call e.g. `main.NewArtifact()` rather than `artifact.New()` etc. For this reason, we're really forced to put them in different packages.

The thing that might make this easiest to understand is if you create a second project which imports your package and uses it. I think as soon as you start trying to call it from outside the package, which is what your library users will also need to do, it will start to make sense.

Reply:
if someone imported the library without giving it a name, e.g. `import ("github.com/taskcluster/taskcluster-lib-artifact-go")`, how would they access the Download() method?  would they have to give it their own name (e.g. `import (artifact "github.com/taskcluster/taskcluster-lib-artifact-go")`) then use artifact?   Are you saying that if I made it `package artifact` that `import ("github.com/taskcluster/taskcluster-lib-artifact-go/artifact")` would allow someone to use `artifact.Download()`.

I think I'm getting this now...  Go modules feel strange to me still

Reply:
Yes that is about right, but with the additional hurdle that when you compile a package called main, it doesn't create a library, it creates a static binary. You can only import libraries, so you wouldn't be able to import something that contained a `main` package. Note, the package represents only one directory level - anything in a subdirectory is a different package - so packages are a little bit different here to other languages. The scope of the package is just the directory it is in.